File: Crawler/Result/Summary/ItemStatus.php
<?php

namespace Crawler\Result\Summary;

enum ItemStatus
{
    case OK;
    case WARNING;
    case ERROR;
    case INFO;

    public static function fromRangeId(int $rangeId): ItemStatus
    {
        return match ($rangeId) {
            0 => ItemStatus::OK,
            1 => ItemStatus::WARNING,
            2 => ItemStatus::ERROR,
            3 => ItemStatus::INFO,
        };
    }

    public static function fromText(string $text): ItemStatus
    {
        return match (strtoupper($text)) {
            'OK' => ItemStatus::OK,
            'WARN' => ItemStatus::WARNING,
            'ERROR' => ItemStatus::ERROR,
            'INFO' => ItemStatus::INFO,
        };
    }

    public static function getSortOrder(ItemStatus $status): int
    {
        return match ($status) {
            self::OK => 3,
            self::WARNING => 2,
            self::ERROR => 1,
            self::INFO => 4
        };
    }
}

File: src/components/summary/item_status.rs
// SiteOne Crawler - Summary ItemStatus
// (c) Jan Reges <jan.reges@siteone.cz>

use serde::{Deserialize, Serialize};

use crate::error::CrawlerError;

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
#[serde(rename_all = "UPPERCASE")]
pub enum ItemStatus {
    Ok,
    Notice,
    Warning,
    Critical,
    Info,
}

impl ItemStatus {
    pub fn from_range_id(range_id: i32) -> Result<Self, CrawlerError> {
        match range_id {
            0 => Ok(ItemStatus::Ok),
            1 => Ok(ItemStatus::Notice),
            2 => Ok(ItemStatus::Warning),
            3 => Ok(ItemStatus::Critical),
            4 => Ok(ItemStatus::Info),
            _ => Err(CrawlerError::Parse(format!(
                "ItemStatus::from_range_id: Unknown range ID '{}'",
                range_id
            ))),
        }
    }

    pub fn from_text(text: &str) -> Result<Self, CrawlerError> {
        match text.to_uppercase().as_str() {
            "OK" => Ok(ItemStatus::Ok),
            "NOTICE" => Ok(ItemStatus::Notice),
            "WARNING" => Ok(ItemStatus::Warning),
            "CRITICAL" => Ok(ItemStatus::Critical),
            "INFO" => Ok(ItemStatus::Info),
            _ => Err(CrawlerError::Parse(format!(
                "ItemStatus::from_text: Unknown status '{}'",
                text
            ))),
        }
    }

    pub fn sort_order(&self) -> i32 {
        match self {
            ItemStatus::Critical => 1,
            ItemStatus::Warning => 2,
            ItemStatus::Notice => 3,
            ItemStatus::Ok => 4,
            ItemStatus::Info => 5,
        }
    }
}
